### DISCUSSION TITLE
Help to setup local dev

### DISCUSSION THREAD

**Discussion by Limezy** [outline/outline#3420] [General] [answered]
Hi, I'm trying to setup a local dev environnement to work on my PR. I am following these instructions : https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/local-development-5hEhFRXow7
Everything goes well except for the authentification. I have setup a Slack auth but it's forbidden to have a "http" URI. Slack requires https, so I can't add `http://localhost:3000/auth/slack.callback` to my URI list. How shall I do ?
Thanks for your help !

PS : I had to set a SMTP username, otherwise everything would crash
PPS : I also had to change the ports for redis and postgresql. Maybe would be interesting to have them at the right value by default in .env.sample

**Comment by tommoor** [comment, marked as answer]
> it's forbidden to have a "http" URI. Slack requires https, so I can't add http://localhost:3000/auth/slack.callback to my URI list. How shall I do ?

Slack only recently added this, and it's dumb but you can just bypass the restriction in their UI by editing the html in the web inspector 🤦 .

> PS : I had to set a SMTP username, otherwise everything would crash
>  I also had to change the ports for redis and postgresql.

We'd accept a PR to fix these i guess!

**Reply by Limezy** [reply]
Wow that's a crazy workaround 🤣 
OK I'll propose a slightly modified .env.sample to ease the work of future devs (not fake devs like me 😅)

**Reply by Limezy** [reply]
It worked ! I now have my local dev setup running, and my PR is working great !

**Comment by almereyda** [comment]
Cool this worked out by working around the client-side form validation within the Slack UI (?) to allow for localhost development endpoints.

In case we wanted to work with this in the future without such a workaround, eventually with other IDPs where this is not possible, a more canonical way to achieve localhost TLS certification is by injecting a self-signed certificate into the system/browser, or by creating a local CA and installing its root certificate, while creating a self-signed leaf certificate for `localhost`.

Useful references to set this up were:

- https://letsencrypt.org/docs/certificates-for-localhost/ Making and trusting your own certificates
  - https://github.com/jsha/minica a small, simple CA intended for use in situations where the CA operator also operates each host where a certificate will be used
- https://words.filippo.io/mkcert-valid-https-certificates-for-localhost/ valid HTTPS certificates for localhost (or for any other name)
- https://github.com/FiloSottile/mkcert A simple zero-config tool to make locally trusted development certificates

These were discovered in course of https://gitlab.com/liberaforms/liberaforms/-/issues/141#note_896572240
